pkg/manifest/manifest.go

```diff
@@ -256,23 +257,33 @@ func (manifest *manifest) RunAsUser() string {
 }
 
 func (manifest *manifest) UnpackAsUser() string {
-	if manifest.ReadOnly {
+	if *manifest.ReadOnly {
 		return "root"
 	}
 
 	return manifest.RunAsUser()
 }
 
 func (manifest *manifest) GetReadOnly() bool {
-	return manifest.ReadOnly
+	if manifest.ReadOnly != nil {
+		return *manifest.ReadOnly
+	}
+
+	return false
+}
+
+func (manifest *manifest) SetReadOnlyIfUnset(readonly bool) {
+	if manifest.ReadOnly == nil {
+		*manifest.ReadOnly = readonly
```

Comment:
if it's not too much difficulty, we try to treat the `manifest` as an immutable representation of what is actually in the pod manifest, I'd rather see wherever the unpacking code is decide what user to use based on pod setting and manifest setting rather than pushing that logic into the manifest